exploratory_analysis/clean_and_split.py

```diff
@@ -0,0 +1,98 @@
+import pandas as pd
+from pathlib import Path
+import numpy as np
+from langdetect import detect
+import logging
+
+RANDOM_SEED = 42
+
+logger = logging.getLogger(__name__)
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+    handlers=[logging.StreamHandler()]
+)
+
+
+def remove_non_english_comments(df: pd.DataFrame) -> pd.DataFrame:
+    logger.info("removing non-english comments")
+    is_english = "is_english"
+    df[is_english] = df.apply(
+        lambda row: detect(row.clean_text) == "en",
+        axis=1,
+    )
+    english_only_df = df[df[is_english]]
+
+    # clean up
+    df.drop(is_english, inplace=True, axis=1)
```

Comment:
No need for this `drop` here since you're just returning `english_only_df`?

Reply:
I want to make sure I return the df originally passed exactly as it was, no sneaky in place changes. 

I checked in the debugger to see if dropping from `df` might also drop from `english_only_df`, but it looks like that df keeps the extra column even if the original that it's slicing from changes. 
<img width="1393" alt="image" src="https://user-images.githubusercontent.com/20362115/204559551-c8eb6f60-6643-42eb-90f6-9018b1fee07d.png">

Need to keep both